client/src/use/useStyling.ts

```diff
@@ -35,82 +55,165 @@ export default function useStyling({ markChangesPending }: UseStylingParams) {
     strokeWidth: 3,
     opacity: 1.0,
     color: 'type',
+    fill: false,
   };
   const selected: Style = {
     ...standard,
     color: vuetify.preset.theme.themes.dark.accent as string,
     strokeWidth: 5,
     opacity: 1.0,
+    fill: false,
   };
   const disabled: Style = {
     ...standard,
     color: 'type',
     strokeWidth: 0.5,
     opacity: 0.45,
+    fill: false,
   };
-  // Colors provided for the different Types
   const stateStyling: StateStyles = { standard, selected, disabled };
-  const typeColors = [
-    colors.red.accent1,
-    colors.yellow.darken3,
-    colors.purple.lighten3,
-    colors.green.lighten3,
-    colors.yellow.lighten3,
-    colors.purple.darken3,
-    colors.green.darken3,
-  ];
-
-  function loadTypeColors(list?: Record<string, string>) {
-    if (list) {
+
+  /**
+   * Generates a color pallette to be used for mapping type names to colors.
+   * It generates colors from a rainbow spectrum and then takes a dark and lighter version
+   * of each color.
+   * @param {int} numColors - number of colors to attempt to generate the higher
+   * the number the more similar the colors will be.  Cyan like colors will be filtered out,
+   * so numColors isn't a guarantee of x*3 (normal, dark, light) colors.
+   */
+  function generateColors(numColors: number) {
+    const colorList = [];
+    for (let i = 0; i < numColors; i += 1) {
+    //We are using a rainbow but we want to skip the cyan area so number will be reduced
+      const pos = (i * (1 / numColors));
+      if (pos > 0.58 && pos < 0.63) {
+        break;
+      }
+      const baseColor = d3.color(d3.interpolateRainbow(pos))?.hex();
+      if (baseColor) {
+        const hueColor = d3.hsl(baseColor);
+        hueColor.s = 1.0;
+        hueColor.l = 0.5;
+        colorList.push(hueColor.hex());
+        hueColor.s = 0.5;
+        hueColor.l = 0.35;
+        colorList.push(hueColor.hex());
+        hueColor.s = 1.0;
+        hueColor.l = 0.75;
+        colorList.push(hueColor.hex());
+      }
+    }
+
+    //Mix up colors in a uniform way so reloads have the same types associated with the same colors
+    let seed = 0.5;
+    colorList.sort(() => {
+      seed += seed;
+      return Math.cos(seed);
+    });
+    return colorList;
+  }
```

Comment:
Then, the reader doesn't have to skim 20 lines of code to figure out that this function can be safely refactored later.